test/sim/misc/random-teams.js

```diff
@@ -181,11 +182,17 @@ describe(`Factory sets`, function () {
 							}
 						}
 
+						assert(!!set.evs, `Set of ${species} has no EVs specified`);
+						const keys = Object.keys(set.evs);
+						const evKeys = ['hp', 'atk', 'def', 'spa', 'spd', 'spe'];
 						let totalEVs = 0;
-						for (const ev in set.evs) {
+						for (const ev of keys) {
+							assert(evKeys.includes(ev), `Invalid EV key (${ev}) on set of ${species}`);
 							totalEVs += set.evs[ev];
-							assert((set.evs[ev] / 4) === Math.floor(set.evs[ev] / 4), `EVs of ${ev} not divisible by 4 on ${species}`);
+							assert.equal(set.evs[ev] % 4, 0, `EVs of ${ev} not divisible by 4 on ${species}`);
 						}
+						const sortedKeys = Utils.sortBy([...keys], ev => evKeys.indexOf(ev));
```

Comment:
Oh hey, you ended up using the thing I recommended; nice!

Reply:
I would probably use `evOrder` rather than `evKeys` to be clearer about what it's for, though.